platform/darwin/darwin.cmake

```diff
@@ -0,0 +1,164 @@
+        ${PROJECT_SOURCE_DIR}/platform/darwin/core/string_nsstring.mm
+        ${PROJECT_SOURCE_DIR}/platform/darwin/core/timer.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/gfx/headless_backend.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/gfx/headless_frontend.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/layermanager/layer_manager.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/map/map_snapshotter.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/platform/time.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/asset_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/mbtiles_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/database_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/file_source_manager.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/file_source_request.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/local_file_request.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/local_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/main_resource_loader.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline_database.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline_download.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/online_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/$<IF:$<BOOL:${MLN_WITH_PMTILES}>,pmtiles_file_source.cpp,pmtiles_file_source_stub.cpp>
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/sqlite3.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/text/bidi.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/compression.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/filesystem.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/monotonic_timer.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/png_writer.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/thread_local.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/utf.cpp
+)
+
+target_include_directories(
+    mbgl-core
+    PUBLIC
+        ${PROJECT_SOURCE_DIR}/platform/darwin/include
+        ${PROJECT_SOURCE_DIR}/platform/default/include
+    PRIVATE
+        ${PROJECT_SOURCE_DIR}/platform/darwin/src ${PROJECT_SOURCE_DIR}/platform/macos/src
+)
+
+if(MLN_WITH_METAL)
+    target_sources(
+        mbgl-core
+        PRIVATE
+            ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/mtl/headless_backend.cpp
+    )
+endif()
+
+include(${PROJECT_SOURCE_DIR}/vendor/icu.cmake)
+
+set(CMAKE_OBJC_FLAGS "-fobjc-arc")
+set(CMAKE_OBJCXX_FLAGS "-fobjc-arc")
```

Comment:
Not sure if this is a good idea. We don't seem to use `-fobjc-arc` for the Bazel builds? 🤔